app/models/job_execution.rb

```diff
@@ -94,6 +99,14 @@ def error!(exception)
   end
 
   def run!
+    begin
+      @start_callbacks.each(&:call)
+    rescue => e
+      @output.write(e.message)
+      finish
+      return
+    end
+
     @output.write('', :started)
```

Comment:
should this come after the `:started` write since that starts the streaming ... might not be sent to the user if it comes before ...